src/aosm/azext_aosm/deploy/pre_deploy.py

```diff
@@ -0,0 +1,366 @@
+        :type artifact_store_type: ArtifactStoreType
+        :param location: The location of the artifact store.
+        :type location: str
+        """
+        logger.info(
+            "Creating artifact store %s if it does not exist",
+            artifact_store_name,
+        )
+        try:
+            self.api_clients.aosm_client.artifact_stores.get(
+                resource_group_name=resource_group_name,
+                publisher_name=publisher_name,
+                artifact_store_name=artifact_store_name,
+            )
+            print(
+                f"Artifact store {artifact_store_name} exists in resource group {resource_group_name}"
+            )
+        except azure_exceptions.ResourceNotFoundError:
+            print(
+                f"Create Artifact Store {artifact_store_name} of type {artifact_store_type}"
+            )
+            poller = (
+                self.api_clients.aosm_client.artifact_stores.begin_create_or_update(
+                    resource_group_name=resource_group_name,
+                    publisher_name=publisher_name,
+                    artifact_store_name=artifact_store_name,
+                    parameters=ArtifactStore(
+                        location=location,
+                        store_type=artifact_store_type,
+                    ),
+                )
+            )
+            # Asking for result waits for provisioning state Succeeded before carrying
+            # on
+            arty: ArtifactStore = poller.result()
+
+            if arty.provisioning_state != PROV_STATE_SUCCEEDED:
+                logger.debug(f"Failed to provision artifact store: {arty.name}")
+                raise RuntimeError(
+                    f"Creation of artifact store proceeded, but the provisioning"
+                    f" state returned is {arty.provisioning_state}. "
+                    f"\nAborting"
+                )
+            logger.debug(
+                f"Provisioning state of {artifact_store_name}"
+                f": {arty.provisioning_state}"
+            )
+
+    def ensure_acr_artifact_store_exists(self) -> None:
+        """
+        Ensures that the ACR Artifact store exists.
+
+        Finds the parameters from self.config
+        """
+        self.ensure_artifact_store_exists(
+            self.config.publisher_resource_group_name,
+            self.config.publisher_name,
+            self.config.acr_artifact_store_name,
+            ArtifactStoreType.AZURE_CONTAINER_REGISTRY,
+            self.config.location,
+        )
+
+    def ensure_sa_artifact_store_exists(self) -> None:
+        """
+        Ensures that the Storage Account Artifact store for VNF exists.
+
+        Finds the parameters from self.config
+        """
+        if not isinstance(self.config, VNFConfiguration):
+            raise AzCLIError(
+                "Check that storage account artifact store exists failed as requires VNFConfiguration file"
```

Comment:
This error is incredibly grammatically confusing. Maybe something like "Check for existence of Storage Account Artifact Store failed because no valid VNFConfiguration was found"?

Reply:
Wow, yes it is confusing! I cannot remember writing it but I'm sure I did. :)